### PR TITLE
doc: add test document

### PR DESCRIPTION
**Pull request by Letty5411**
Signed-off-by: letty.ll <letty.ll@alibaba-inc.com>

**1.Describe what this PR did**
Add doc about how to run pouch test.
Developing test part will be added later after more discussion.
**2.Does this pull request fix one issue?** 

**3.Describe how you did it**

**4.Describe how to verify it**

**5.Special notes for reviews**

### CODE REVIEW

File: docs/test.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
+# Introduction
+
+In order to provide high quality `pouch`, tests are an important part.
```
Comment (sunyuan3) [inline docs/test.md]: s/tests are/testing is/
"are an" is invalid.

File: docs/test.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+* run entire test: 
+
+	```
+	#go test ./test
+	```
+* run a single test suite: 
+
+	```
+	#go test -check.f PouchHelpSuite
+	OK: 1 passed
```
Comment (sunyuan3) [inline docs/test.md]: "1 passed" is not good. Please use another _test.go file as an example and make sure the file contain at least 2 cases.

File: docs/test/test.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+*  the usage of pouch test
+*  the development of pouch test
+
+# Organization of test
+
+Test in pouch could be divided into two parts [unit testing](https://en.wikipedia.org/wiki/Unit_testing#Description) and [integration testing](https://en.wikipedia.org/wiki/Integration_testing). More info could be get from the wiki page. For pouch developer, if your code is only used in a single module, then the unit test is enough. While if your code is called by multiple modules, integration tests are required. In pouch, both of them are developed using go language. More details could be got in [Unit Testing](#unit-test) and [Integration Testing](#intg-test).
+
+# Usage of Test
+
+## <a name="unit-test"></a> Unit Testing
```
Comment (xiang90) [inline docs/test/test.md]: use markdown instead of html tag?

File: docs/test/test.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+
```
Comment (allencloud) [inline docs/test/test.md]: I think the first blank line is unnecessary.

File: docs/test/test.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+# Organization of test
+
+Test in pouch could be divided into two parts [unit testing](https://en.wikipedia.org/wiki/Unit_testing#Description) and [integration testing](https://en.wikipedia.org/wiki/Integration_testing). More info could be get from the wiki page. For pouch developer, if your code is only used in a single module, then the unit test is enough. While if your code is called by multiple modules, integration tests are required. In pouch, both of them are developed using go language. More details could be got in [Unit Testing](#unit-test) and [Integration Testing](#intg-test).
+
+# Usage of Test
+
+## <a name="unit-test"></a> Unit Testing
+
+Unit testing uses [go testing](https://golang.org/pkg/testing/) package, named with `_test.go` postfix and always locates in the same directory with the code tested. `client/client_test.go` is a good example of unit test.
```
Comment (allencloud) [inline docs/test/test.md]: I suggest that we could provide a hyperlink for `client/client_test.go`, then readers can easily go to that file directly and get what they want if they like.

File: docs/test/test.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+There are two ways to trigger unit test, from the makefile target to run entire unit test or just using `go test $testdir` to run unit test in a specified directory. Here is an example:
```
Comment (allencloud) [inline docs/test/test.md]: s/makefile/Makefile
and also provide a hyperlink for work `Makefile`:

There are two ways to trigger unit test. First, Use [Makefile](https://github.com/alibaba/pouch/blob/master/Makefile) target `unit-test` to run entire unit test, and the command is `make unit-test` in the root path of pouch. Second, use `go test $testdir` to run unit test in a specified directory. Here is an example:

File: docs/test/test.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+Integration test is in `pouch/test`, programmed with `go language`. There are two kinds of integration test, API test named as `pouch_api_xxx_test.go` and command line test named as `pouch_cli_xxx_test.go` ("xxx" represents the test point). 
+
+It uses [go-check](https://labix.org/gocheck) package, a rich testing framework for go language. It provides many useful functions, such as:
+
+* SetUpTest: Run before each test to do some common work.
+* TearDownTest: Run after each test to do some cleanup work.
+* SetUpSuite: Run before each suite to do common work for the whole suite.
+* TearDownSuite: Run after each suite to do cleanup work for the whole suite.
+
```
Comment (allencloud) [inline docs/test/test.md]: Here we have a redundant blank line. 😄

File: docs/test/test.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+	PASS
+	ok  	github.com/alibaba/pouch/test	3.081s
+	```
+* run a single test case:
+
+	```
+	#go test -check.f PouchHelpSuite.TestHelpWorks
+	OK: 1 passed
+	PASS
+	ok  	github.com/alibaba/pouch/test	0.488s	```
```
Comment (allencloud) [inline docs/test/test.md]: The format is somewhat incorrect. ``` should be in a new line with a blank line followed.

**Comment by allencloud** [comment]
I have left some comments on syntax. While I am wondering if we should first make sure user's environment can run the test directory. At my first glance, unit test seems OK. But integration needs more strict environment, for instance, my develop machine is MACOS and it is not support to run integration test. In addition, I am not sure if users can easily run integration test on there Linux machine. @Letty5411

**Comment by Letty5411** [comment]
@allencloud Yes, this is a big problem, that users need to configure their test ENV. I should also add this part!

**Comment by Letty5411** [comment]
@allencloud I think users may need to run "make install" to install pouch in their env, then integration test could be run.

**Comment by allencloud** [comment]
> users may need to run "make install" to install pouch in their env, then integration test could be run.
Let's add that !

File: docs/test/test.md (hunk 4 above)
Comment (Letty5411) [reply docs/test/test.md]: ok

File: docs/test/test.md (hunk 3 above)
Comment (Letty5411) [reply docs/test/test.md]: ok

**Comment by allencloud** [comment]
LGTM, I suggest a follow-up with `make install` could added soon. Otherwise, users cannot do testing thing on their local machine.

**Pull request merged**